src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,32 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.InvalidDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int LOGIN_PASS_LENGTH = 6;
+    private static final int ACCEPTABLE_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidDataException("User is null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidDataException("User already exists");
+        }
+        if (user.getLogin().length() < LOGIN_PASS_LENGTH) {
+            throw new InvalidDataException("Login must be at least 6 characters length");
+        }
+        if (user.getPassword().length() < LOGIN_PASS_LENGTH) {
+            throw new InvalidDataException("Password must be at least 6 characters length");
+        }
+        if (user.getAge() < ACCEPTABLE_AGE) {
+            throw new InvalidDataException("Age must be at least 18");
```

Comment:
don't hardcode, you have constants for this. This code is fragile - for example, if you later on change min length to 8, you may forget to change exception messages, and it will confuse users